### PR TITLE
Feature: RAI feeder pool

### PR DESCRIPTION
**Pull request by dimsome**
Todo: 

- [x]  Remove APY from deprecated Pool
- [x] Check if this works with new TheGraph deploy
- [x] FEI is hidden
- [x] FEI is hidden in token list (for e.g. Save)
- [x] RAI Save deposit/withdrawal conversion works as intended
- [x] Swap distance calculation for RAI

### CODE REVIEW

File: apps/protocol/src/app/hooks/useMassetExchangeRate.ts [hunk 1]
```diff
@@ -52,45 +52,38 @@ export const useExchangeRateForMassetInputs = (
...
 
-    const totalAmount = Object.values(touched).reduce((prev, v) => prev.add(scaleAssetValue(v)), BigDecimal.ZERO)
+      let scaledAmount = amount.mulRatioTruncate(feederPool.fasset.ratio).setDecimals(18)
 
-    if (totalAmount) {
-      if (estimatedOutputAmount.value.exact.eq(0) || totalAmount.exact.eq(0)) {
-        return { error: 'Output amount must be greater than zero' }
+      if (feederPool.fasset.token.symbol === 'RAI') {
+        // For RAI, scale back down; the ratio is the relative price :)
+        scaledAmount = feederPool.fasset.price.divPrecisely(scaledAmount)
```
Comment (JamesLefrere) [inline apps/protocol/src/app/hooks/useMassetExchangeRate.ts:76]: The ratio is `302774230` to scale it to masset size. For assets with different decimals, we still need the first step, but here, we need to undo that (retaining the correct units)

File: apps/protocol/src/app/hooks/useScaledInput.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    inputRatios[address] ? amount.mulRatioTruncate(inputRatios[address]).setDecimals(18) : amount,
+  )
+  const scaledHighTotal = BigDecimal.sum(...scaledHighAmounts)
+
+  return touched.reduce<ScaledInput>(
+    (prev, { amount: high, address }, idx) => {
+      const scaledHigh = scaledHighAmounts[idx]
+
+      const low = high.divPrecisely(highTotal)
+      const lowTotal = prev.lowTotal.add(low)
```
Comment (JamesLefrere) [inline apps/protocol/src/app/hooks/useScaledInput.ts:23]: PS @rafaelugolini if it's not clear, the low amounts are used to calculate the price impact; high == the amount the user has selected, low == a small amount of the same proportion. If the distance is quite different, that's price impact.

File: apps/protocol/src/app/pages/Pools/Detail/EthereumPool.tsx [hunk 3]
```diff
@@ -156,11 +156,12 @@ const PoolDetailContent: FC = () => {
   const color = assetColorMapping[title]
   const isLowLiquidity = massetPrice ? liquidity.simple * (massetPrice.value ?? 0) < 100000 : false
 
+  // FIXME REVERT!
```
Comment (JamesLefrere) [inline apps/protocol/src/app/pages/Pools/Detail/EthereumPool.tsx:159]: Found this, fixing :)

File: libs/base/src/lib/context/NetworkProvider.tsx [hunk 4]
```diff
@@ -238,9 +238,9 @@ const ETH_MAINNET: EthereumMainnet = {
     merkleDrop: [graphHostedEndpoint('mstable', 'mstable-merkle-drop')],
     snapshot: ['https://hub.snapshot.org/graphql'],
     feeders: [
-      graphMainnetEndpoint('0x021c1a1ce318e7b4545f6280b248062592b71706', 0, process.env.NX_FEEDERS_SUBGRAPH_API_KEY as string),
       // Temporary preview URL because indexers haven't picked up the new version...
-      'https://api.studio.thegraph.com/query/948/mstable-feeder-pools-and-vaults/v0.0.9',
+      'https://api.studio.thegraph.com/query/948/mstable-feeder-pools-and-vaults/v0.0.11',
```
Comment (JamesLefrere) [inline libs/base/src/lib/context/NetworkProvider.tsx:242]: We will revert the order once there are indexers. The deploy tx hasn't been made yet, we need this in production first.

File: apps/protocol/src/app/hooks/useScaledInput.ts (hunk 2 above)
Comment (JamesLefrere) [reply apps/protocol/src/app/hooks/useScaledInput.ts:23]: The tests should make this clear, I hope 🤞

File: libs/hooks/src/lib/useBigDecimalInputs.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { BigNumber } from 'ethers'
```
Comment (rafaelugolini) [inline libs/hooks/src/lib/useBigDecimalInputs.ts:1]: @JamesLefrere do we need this here?

File: apps/protocol/src/app/hooks/useEstimatedOutputMulti.ts [hunk 6]
```diff
@@ -37,52 +36,48 @@ interface Output {
...
+    )
 
     return { value }
-  }, [estimatedOutputRange.fetching, estimatedOutputRange.value, touched, route, lpPriceAdjustment, massetConfig.lowInputValue])
+  }, [estimatedOutputRange.fetching, estimatedOutputRange.value, scaledInput, lpPriceAdjustment, route])
 
   const [update] = useDebounce(
     () => {
-      if (!contract || !touched.length) return {}
+      if (!contract || Object.values(scaledInput.values).length === 0) return {}
```
Comment (rafaelugolini) [inline apps/protocol/src/app/hooks/useEstimatedOutputMulti.ts:63]: @JamesLefrere `scaledInput.values` can be `undefined`

Comment (JamesLefrere) [reply apps/protocol/src/app/hooks/useEstimatedOutputMulti.ts:63]: It shouldn't be, but you made me check and found that it wasn't initialised in the `reduce` 👍

File: libs/bigdecimal/src/lib/BigDecimal.ts [hunk 7]
```diff
@@ -14,6 +14,10 @@ export class BigDecimal {
 
   static ONE = new BigDecimal((1e18).toString())
 
+  static sum(...values: BigDecimal[]): BigDecimal {
+    return values.reduce((prev, current) => prev.add(current), BigDecimal.ZERO)
+  }
```
Comment (chrisjgf) [inline libs/bigdecimal/src/lib/BigDecimal.ts:19]: Such a game changer ❤️

**Pull request merged**